### PR TITLE
Fix/map execption semantically correct

### PR DESCRIPTION
**Pull request by HaMatthias**
Hi, alongside with the docker-compose pull request #1528 I did an exception handling for requesting non existing port mappings.
I am very new to programming so feel free for hints/mistakes.

Please consider the other PR as well.

### CODE REVIEW

**Comment by bsideup** [comment]
@HaMatthias could you please update your PR? It is based on a merged branch and has some conflicts now since the branch was merged

File: core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java [hunk 1]
```diff
@@ -447,7 +447,15 @@ public String getServiceHost(String serviceName, Integer servicePort) {
      * @return a port that can be used for accessing the service container.
      */
     public Integer getServicePort(String serviceName, Integer servicePort) {
-        return ambassadorContainer.getMappedPort(ambassadorPortMappings.get(getServiceInstanceName(serviceName)).get(servicePort));
+        Map<Integer, Integer> portMap = ambassadorPortMappings.get(getServiceInstanceName(serviceName));
+
+        if(portMap == null) {
```
Comment (rnorth) [inline core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java]:
```suggestion
        if (portMap == null) {
```

File: core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java [hunk 2]
```diff
@@ -447,7 +447,15 @@ public String getServiceHost(String serviceName, Integer servicePort) {
...
+            throw new IllegalArgumentException("\n\nContext: Getting the port of '" + serviceName + "'.\n" +
```
Comment (rnorth) [inline core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java]: I feel a bit uncomfortable with newlines in exception messages, especially at the beginning. Some proposed rewording:

```suggestion
            throw new IllegalArgumentException("Could not get a port for '" + serviceName + "'. " +
```

File: core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java [hunk 3]
```diff
@@ -447,7 +447,15 @@ public String getServiceHost(String serviceName, Integer servicePort) {
...
+                "Problem: '" + serviceName + "' does not exist.\n" +
```
Comment (rnorth) [inline core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java]:
```suggestion
                "Reason: '" + serviceName + "' does not exist. " +
```

File: core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java [hunk 4]
```diff
@@ -447,7 +447,15 @@ public String getServiceHost(String serviceName, Integer servicePort) {
...
+                "Solution: Expose the service '" + serviceName + "' at the container setup of this instance.");
```
Comment (rnorth) [inline core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java]:
```suggestion
                "To fix, please ensure that the service '" + serviceName + "' has ports exposed using .withExposedService(...)");
```

**Comment by rnorth** [comment]
@HaMatthias please could you have a look at my suggested changes?

**Comment by rnorth** [comment]
@HaMatthias I think you may have forgotten to push! Please could you check?

**Comment by rnorth** [comment]
@HaMatthias I see you've merged master back into your branch, but it still doesn't look like you've pushed any changes since my review. It looks like you've resolved the comments without changes:

image

Please could you push, if you've made changes? Also, please allow us to push the resolve button after re-reviewing.

File: core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java [hunk 5]
```diff
@@ -447,7 +447,15 @@ public String getServiceHost(String serviceName, Integer servicePort) {
      * @return a port that can be used for accessing the service container.
      */
     public Integer getServicePort(String serviceName, Integer servicePort) {
-        return ambassadorContainer.getMappedPort(ambassadorPortMappings.get(getServiceInstanceName(serviceName)).get(servicePort));
+        Map<Integer, Integer> portMap = ambassadorPortMappings.get(getServiceInstanceName(serviceName));
+
+        if (portMap == null) {
+            throw new IllegalArgumentException("Could not get a port for '" + serviceName + "'. " +
+                "Reason: '" + serviceName + "' does not exist. " +
```
Comment (rnorth) [inline core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java]: Sorry, one more tiny tweak. I'll commit this myself.
```suggestion
                "Testcontainers does not have an exposed port configured for '" + serviceName + "'. +
```

Comment (HaMatthias) [reply core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java]: No worries, sorry for late committing. I do not find the settings for this. Can you help me here?

**Comment by HaMatthias** [comment]
I do not know were I can allow you

**Comment by rnorth** [comment]
🤦‍♂ sorry, it looks like my suggested change had an unclosed quote 😱

**Comment by rnorth** [comment]
I need to close this PR to replace it with a new, fixed branch.

**Comment by rnorth** [comment]
Thank you for the contribution @HaMatthias - finally merged in #2154!

**Pull request closed without merging**